src/style/style.js

```diff
@@ -1417,6 +1439,71 @@ class Style extends Evented {
         }
 
         this._updateDrapeFirstLayers();
+        this._markersNeedUpdate = true;
+    }
+
+    _createFog(fogOptions: FogSpecification) {
+        const fog = this.fog = new Fog(fogOptions);
+        this.stylesheet.fog = fogOptions;
+        const parameters = {
+            now: browser.now(),
+            transition: extend({
+                duration: 0
+            }, this.stylesheet.transition)
+        };
+
+        fog.updateTransitions(parameters);
+    }
+
+    _updateMarkersOpacity() {
```

Comment:
Adding marker rendering updates to the style breaks the separation between DOM-based and non-DOM based map elements. I see that `Marker._update` is called on all camera changes. Can this update be handled entirely in the Marker class using an event instead?

Reply:
That's a good point, I think we can directly use the DOM queue recently introduced in https://github.com/mapbox/mapbox-gl-js/pull/10530 for that purpose.